resources/run_adl.sh

```diff
@@ -0,0 +1,158 @@
+  else 
+    return 1
+  fi
+  
+}
+
+check_mongod() {
+  MONGOD="mongod"
+  
+  check_procname $MONGOD
+  process_check_result=$?
+  check_port $MONGOD_PORT
+  port_check_result=$?
+
+  if [[ $process_check_result -eq 0 ]] && [[ $port_check_result -eq 0 ]]; then
+    return 0
+  else
+    return 1
+  fi
+}
+
+check_mongohoused() {
+  MONGOHOUSED="mongohoused"
+  
+  check_procname $MONGOHOUSED
+  process_check_result=$?
+  check_port $MONGOHOUSED_PORT
+  port_check_result=$?
+
+  if [[ $process_check_result -eq 0 ]] && [[ $port_check_result -eq 0 ]]; then
+    return 0
+  else
+    return 1
+  fi
+}
+
+# check if mac or linux
+if [ $OS = "Linux" ]; then
+  distro=$(awk -F= '/^NAME/{print $2}' /etc/os-release)
+  if [ "$distro" = "\"Red Hat Enterprise Linux\"" ] ||
+[ "$distro" = "\"Red Hat Enterprise Linux Server\"" ]; then
+    export VARIANT=rhel7
+    MONGO_DOWNLOAD_LINK=$MONGO_DOWNLOAD_REDHAT
+  elif [ "$distro" = "\"Ubuntu\"" ]; then
+    export VARIANT=ubuntu1804  
+    MONGO_DOWNLOAD_LINK=$MONGO_DOWNLOAD_UBUNTU
+  else
+    echo ${distro} not supported
+    exit 1
+  fi
+elif [ $OS = "Darwin" ]; then
+  MONGO_DOWNLOAD_LINK=$MONGO_DOWNLOAD_MAC
+else
+  echo $(uname) not supported
+  exit 1
+fi
+
+check_mongod
+if [[ $? -ne 0 ]]; then
+  # Install and start mongod
+  if [ $OS = "Linux" ]; then
+    curl -O $MONGO_DOWNLOAD_BASE/linux/$MONGO_DOWNLOAD_LINK
+  else
+    curl -O $MONGO_DOWNLOAD_BASE/osx/$MONGO_DOWNLOAD_LINK
+  fi
+  
+  tar zxvf $MONGO_DOWNLOAD_LINK
+  
+  mkdir -p $MONGO_DB_PATH
+  mkdir -p $MONGO_LOGS_PATH
+
+  # Note: ADL has a storage.json file that generates configs for us. The mongodb source is on port 28017 so we use that here.
+  ${MONGO_DOWNLOAD_LINK:0:$((${#MONGO_DOWNLOAD_LINK} - 4))}/bin/mongod --port $MONGOD_PORT --dbpath $MONGO_DB_PATH --logpath $MONGO_LOGS_PATH/test.log --fork
+fi
+
+check_mongohoused
+if [[ $? -ne 0 ]]; then
+  # Install and start mongohoused 
+  git config --global url.git@github.com:.insteadOf https://github.com/
+  # Clone the mongohouse repo
+  if [ ! -d "$MONGOHOUSE_DIR" ]; then
+    git clone $MONGOHOUSE_URI
+  fi
+  cd $MONGOHOUSE_DIR
+  git pull $MONGOHOUSE_URI
+  
+  OS_LOWERCASE=$(echo $OS | tr '[:upper:]' '[:lower:]')
+  curl -O "https://dl.google.com/go/${GOVERSION}.${OS_LOWERCASE}-amd64.tar.gz"
+  mkdir $GO_DIR
+  tar -C $GO_DIR -xzf ${GOVERSION}.${OS_LOWERCASE}-amd64.tar.gz
+  export PATH=$GO_DIR/go/bin:$PATH
+  # May have been set from previous install
+  unset GOROOT
+  
+  go mod vendor
+
+  # Download external dependencies
+  go run cmd/buildscript/build.go tools:download:mqlrun
+  go run cmd/buildscript/build.go tools:download:mongosql
+  
+  # Set relevant environment variables
+  export GOPRIVATE=github.com/10gen
+  export MONGOHOUSE_ENVIRONMENT="local"
+  export MONGOHOUSE_MQLRUN="$(pwd)/artifacts/mqlrun"
+  export LIBRARY_PATH="$(pwd)/artifacts"
+  
+  # Load tenant config into mongodb
+  # TODO - we should probably make a different tenant config than the one that exists at
```

Comment:
This may take some effort, and may not be necessary.  I propose I create a ticket for this and address it in a future release if needed.

Reply:
You're right.  I tried querying the underlying mongod data and it turns out I needed to update the config to add the storage/stores configuration.  I'll see what is a good way to add that config, perhaps there's a straightforward way to add to the existing config on ADL.

Reply:
Yea we could possibly make use of something known as "wildcard" datasources in ADL so we could query arbitrary mongodb data. I don't know the full details off-hand but some of the docs (like [these](https://docs.mongodb.com/datalake/admin/manage-namespace-catalog-cli/)) may help, or we could ask the ADL team for guidance.

One idea is we could copy the config file we're currently using from the mongohouse repo and just include it in our repo in the resources folder. We could remove all the extra datasources we won't use and just add a wildcard mongodb source.

Reply:
Added a wildcard collection for the tenant-config.json.  It uses the 'jq' command line tool, which I confirmed is on Ubuntu and RHEL.  I don't have access to the macOS host yet to check.
The script deletes the existing `storage` section and then adds the wildcard collection for the database: integration_test.  I confirmed that is the only database present on the local ADL instance.